botorch/optim/optimize.py

```diff
@@ -1177,6 +1240,86 @@ def _gen_batch_initial_conditions_local_search(
     raise RuntimeError(f"Failed to generate at least {min_points} initial conditions")
 
 
+def _gen_starting_points_local_search(
+    discrete_choices: list[Tensor],
+    raw_samples: int,
+    X_avoid: Tensor,
+    inequality_constraints: list[tuple[Tensor, Tensor, float]],
+    min_points: int,
+    acq_function: AcquisitionFunction,
+    max_batch_size: int = 2048,
+    max_tries: int = 100,
+):
+    X_init = _gen_batch_initial_conditions_local_search(
+        discrete_choices=discrete_choices,
+        raw_samples=raw_samples,
+        X_avoid=X_avoid,
+        inequality_constraints=inequality_constraints,
+        min_points=min_points,
+        max_tries=max_tries,
+    )
+
+    # pick the best starting points
+    with torch.no_grad():
+        acqvals_init = _split_batch_eval_acqf(
+            acq_function=acq_function,
+            X=X_init.unsqueeze(1),
+            max_batch_size=max_batch_size,
+        ).unsqueeze(-1)
+
+    return X_init[acqvals_init.topk(k=min_points, largest=True, dim=0).indices]
+
+
+def _gen_first_candidate_starting_points_local_search(
```

Comment:
It seems like you could simplify this by just having a single funciton `_gen_starting_points_local_search` and just check whether `batch_initial_conditions` is provided and branch on that? Then you can call that recursively with  `batch_initial_conditions=None` in that funciton.